core/domain/exp_domain.py

```diff
@@ -873,9 +873,6 @@ def validate(self, exp_param_specs_dict):
             if self.solution:
                 Solution.from_dict(
                     self.id, self.solution).validate(self.id)
-            else:
```

Comment:
Currently, we're assuming that Solution is not mandatory if a Hint is specified. This is why I had to remove that check.